pytorch/training/docker/1.12/py3/cu113/Dockerfile.gpu

```diff
@@ -515,6 +515,16 @@ RUN  wget -nv https://github.com/rapidsai/rmm/archive/v${RMM_VERSION}.tar.gz \
 # Install ZeRO-2D Binary
 RUN pip install --no-cache-dir -U ${ZERO_2D_URL}
 
+# Install flash attention
+WORKDIR /flash
+RUN git clone https://github.com/HazyResearch/flash-attention.git \
+    && cd flash-attention \
+    && git checkout 1d0b41be3b0fa759502e274779db6758617fc7b9 \
```

Comment:
nit: do we need to parameterize/reflect as an env var the commit id? We want to make sure this is reviewed once the dockerfile changes are carried forward to new versions.

Reply:
Changing this to an ARG defined at the top of the dockerfile, alongside the framework binaries, and adding a comment with reasoning.

Reply:
I haven't got the reply from team yet, but the rubik team changes mirror that in https://code.amazon.com/packages/Rubik/blobs/tp_v2/--/containers/dockerfiles/Dockerfile.modelparallel.pytorch1.12.

Also the `ARG SMD_MODEL_PARALLEL_URL` points to the latest green rubik including those changes, so should have been validated there.

Reply:
Thanks @sheilaliuxl! Will add a placeholder for now.

Reply:
A quick update: Flash attention did not have releases until recently, so we have to get the commit id for now.
In the next release, we can use their release tags if present.

Reply:
Hi, I sent another PR: https://github.com/ShiboXing/deep-learning-containers/pull/4

This should fix the flash attention test. Note that most changes are from sync-to-head, and the only change is [this commit](https://github.com/ShiboXing/deep-learning-containers/commit/85dbac227580faa0138591f862513c2b35467fc1), thanks a lot for all your help!